zebra/zebra_nhg.c

```diff
@@ -2648,7 +2648,7 @@ static unsigned nexthop_active_check(struct route_node *rn,
 
 		ifp = if_lookup_by_index(nexthop->ifindex, nexthop->vrf_id);
 
-		if (ifp && ifp->vrf->vrf_id == vrf_id && if_is_up(ifp)) {
+		if (ifp && ifp->vrf->vrf_id == vrf_id && if_is_up(ifp) && if_connected_count(ifp->connected)) {
 			SET_FLAG(nexthop->flags, NEXTHOP_FLAG_ACTIVE);
```

Comment:
I wonder if it is enough. what happens if, for a given ipv4 route, you remove all ipv4 addresses, knowing that you have 1 ipv6 address left? or the opposite for a given ipv6 route.

should not we involve the 'v6-with-v4-nexthop' flag into this control too ?